tests/test_visitors/test_tokenize/test_comments/test_comment_in_formatted_string312.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+
+from wemake_python_styleguide.compat.constants import PY312
+from wemake_python_styleguide.violations.best_practices import (
+    CommentInFormattedStringViolation,
+)
+from wemake_python_styleguide.visitors.tokenize.comments import (
+    CommentInFormattedStringVisitor,
+)
+
+if not PY312:  # pragma: >=3.12 no cover
+    pytest.skip(
+        reason='comments in fstring was added in 3.12', allow_module_level=True
+    )
+
+# Correct
+fstring_without_comments = """
+foo = f"test{a}"
+"""
+
+fstring_with_hash_single_quotes = """
+foo = f'test {a} # testing'
+"""
+
+fstring_with_hash = """
+foo = f"test{a} # testing"
+"""
+
+fstring_with_hash_between_braces = """
+foo = f"test{a} # comment {b} # comment"
+"""
+
+
+fstring_with_two_values = """
+f"My name is {name} and I am {age} years old."
+"""
+
+
+fstring_with_math_operation = """
+f"The sum of {x} and {y} is {x + y}."
+"""
+
+fstring_with_hash_between_quotes = """
+foo = f"hello" # comment "{bar}" # comment"
+"""
+
+
+fstring_in_docstring = '''
+foo = f"""hello"{bar}" # comment world"""
+'''
+
+# Wrong
+fstring_with_comment = """
+foo = f"test{a # comment
+}"
+"""
+
+fstring_with_comment_single_quotes = """
+foo = f'test{a # comment
+}'
+"""
+
+fstring_with_comment_and_hash_on_new_line = """
+foo = f"test{a # comment
+} # comment"
+"""
+
+fstring_with_two_values_and_comment = """
+foo = f"test{a} and test{b # Testing values
+}"
+"""
+
+
+multiline_fstring_with_comment = """
+foo = (f"test{value}"
+       f"test{another_value}"
+       f"test{wrong # This is not allowed
+       }"
+       f"test{value}")
+"""
+
+fstring_with_comment_and_second_line = """
+foo = f"hello{bar # comment
+}world"
+"""
+
+fstring_with_comment_between_quotes = """
+foo = f"hello'{bar # comment
+}'world"
+"""
+
+fstring_with_comment_in_docstring = '''
```

Comment:
please, add wrong examples for `fr` and `rf` cases with `'` and `"` and `'''` and `"""`